File: April_2022/03_04_2022/miscommunication_from_mars.py
n = int(input())
data = [int(input()) for i in range(n)]
d = int(input())
xorr = int(input())

ans = []
for i in range(len(data)):
    for j in range(len(data)):
        if(i != j and abs(data[i]-data[j]) == d and data[i]^data[j] == xorr):
            x = sorted([data[i],data[j]])
            if(x not in ans):
                ans.append(x)
for i in ans:
    print(i[0],i[1],sep="\t")

File: April_2022/03_04_2022/miscommunication_from_mars.cpp
#include <iostream>
using namespace std;

int getXOR(int x, int y) {
  return (x | y) & (~x | ~y);
}

bool check(int arr[][2],int index,int x,int y){
    for(int i=0;i<index;i++){
        if(arr[i][0] == x && arr[i][1] == y){
            return false;
        }
    }
    return true;
}


int main(){

    int n;
    cin>>n;

    int data[n];
    for(int i=0;i<n;i++){
        cin>>data[i];
    }

    int d,xorr;
    cin>>d>>xorr;

    int arr[n][2];
    int index = 0;

    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if(i!=j && abs(data[i]-data[j]) == d && getXOR(data[i],data[j]) == xorr){
                int x = min(data[i],data[j]);
                int y = max(data[i],data[j]);
                if(check(arr,index,x,y)){
                    arr[index][0] = x;
                    arr[index++][1] = y;
                }
            }
        }
    }

    for(int i=0;i<index;i++){
        printf("%d\t%d\n",arr[i][0],arr[i][1]);
    }

    return 0;
}